kivy/core/text/__init__.py

```diff
@@ -604,7 +605,16 @@ def render(self, real=False):
         if strip:
             text = text.strip()
         if uw is not None and options['shorten']:
-            text = self.shorten(text)
+            short_text = self.shorten(text)
+            chr = type(text)
+            flat_text = text.replace(chr('\n'), chr(' '))
+            if flat_text != short_text:
```

Comment:
This seems expensive as it's done twice, once during the shorten and again for testing.